presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/ReorderJoins.java

```diff
@@ -98,23 +99,22 @@ public ReorderJoins(CostComparator costComparator)
         }
 
         MultiJoinNode multiJoinNode = toMultiJoinNode(joinNode, lookup);
-        try {
-            return Optional.of(new JoinEnumerator(idAllocator, symbolAllocator, session, lookup, multiJoinNode.getFilter(), costComparator).chooseJoinOrder(multiJoinNode.getSources(), multiJoinNode.getOutputSymbols()));
-        }
-        catch (JoinEnumerator.UnknownCostException unknownCostException) {
-            return Optional.empty();
-        }
+        JoinEnumerationResult result = new JoinEnumerator(idAllocator, symbolAllocator, session, lookup, multiJoinNode.getFilter(), costComparator).chooseJoinOrder(multiJoinNode.getSources(), multiJoinNode.getOutputSymbols());
```

Comment:
@findepi I implemented a version that doesn't use exceptions.  I think its messy and bug prone because you need to deal with calculating and handling unknown_cost in a lot of places, so I'd prefer to stick with the exception approach.  but let me know what you think.